lib/Tooling/Refactor/RenameIndexedFile.cpp

```diff
@@ -81,7 +81,9 @@ static MatchKind checkOccurrence(const IndexedOccurrence &Occurrence,
   if (BeginLoc.isInvalid())
     return MatchKind::None;
   StringRef SymbolNameStart = Symbol.Name[0];
-  SourceLocation EndLoc = BeginLoc.getLocWithOffset(SymbolNameStart.size());
+  SourceLocation EndLoc = BeginLoc.getLocWithOffset(std::max(
+      SymbolNameStart.size(),
+      size_t(1))); // Take empty Objective-C selector pieces into account.
   if (!SM.isBeforeInTranslationUnit(BeginLoc, EndLoc)) {
```

Comment:
Ah. Right. It looks like we don't need this check at all now. The initial implementation used `Lexer::getSourceText(
      CharSourceRange::getCharRange(BeginLoc, EndLoc)` so we needed to verify the range, but now we just lex the token so this is redundant.